docs/guides/cloudflare.mdx

```diff
@@ -66,68 +97,136 @@ const getData = async () => {
 
 ```ts
+import { env } from 'cloudflare:workers'; // eslint-disable-line import/no-unresolved
+
 const get404Html = async () => {
-  const c = getHonoContext<{ Bindings: Env }>();
-  return c.env.ASSETS
-    ? await (await c.env.ASSETS.fetch('https://example.com/404.html')).text()
+  return env.ASSETS
+    ? await (await env.ASSETS.fetch('https://example.com/404.html')).text()
     : '';
 };
 ```
 
 Note that `ASSETS.fetch` requires a fully qualified URL, but the origin is ignored. You can use `https://example.com` or any valid origin. It is just an internal request.
 
+It is also possible to always run the worker before serving static assets. See the documentation for [`run_worker_first`](https://developers.cloudflare.com/workers/static-assets/binding/#run_worker_first).
+
 ### Custom Headers
 
 You can set response headers on the `res` object of the Waku context. Since Cloudflare supports response body streaming, server components might not be able to set headers if they were already sent in the response stream. Headers can be set from custom Waku middleware.
 
-Cloudflare released static assets for Workers in the fall of 2024. Cloudflare Workers static assets do not yet support custom headers for the static assets. If that is required, then the file must not be included in the dist/assets folder and instead served via the worker through the ASSETS binding which is accessible on the Hono context.
+For static assets, add a [`_headers`](https://developers.cloudflare.com/workers/static-assets/headers/#custom-headers) file to the root of your `public` folder to set custom headers for static assets.
 
-## Service Bindings and Smart Placement
+> An example `_headers` file is included in Waku's starter template for Cloudflare Workers to prevent indexing of RSC files by search engines:
 
-Cloudflare lets you deploy other Cloudflare Workers functions to your account and connect to them from the Waku worker via ["service bindings"](https://developers.cloudflare.com/workers/runtime-apis/bindings/service-bindings/).
+`./public/_headers`:
 
-This can be combined with [smart placement](https://developers.cloudflare.com/workers/configuration/smart-placement/) and React Suspense to process certain operations in a worker co-located with a D1 database while the main request is handled via in the Workers function running at global edge location. Note that you want to enable smart placement enabled for your Cloudflare Workers function with the D1 binding but have it disabled for your Workers function that is handling requests. You can move interaction with D1 and other co-located resources to the worker and then call it from your server components via the service binding for optimal performance.
-
-## Extending Hono
+```txt
+/RSC/*
+  X-Robots-Tag: noindex
+```
 
-While Waku uses Hono as an HTTP server adapter, Waku also has its own middleware system and custom middleware can easily be added to your `waku.config.ts` file.
+### Static Apps Without A Worker
```

Comment:
This section look good so far.

re: https://github.com/wakujs/waku/pull/1795#issuecomment-3693733809

We could reconsider about the examples, but not now.